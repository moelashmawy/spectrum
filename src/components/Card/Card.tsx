import React from "react";
import styled from "styled-components";
import Button from "../Button/Button";

const StyledInfoSection = styled.div<{ $isAlert?: boolean }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 20%;
  height: 300px;
  background: rgba(51, 170, 51, 0.1);
  border-radius: 15px;
  padding: 20px;
  animation: ${(props) => props.$isAlert && "blinker 1s linear infinite"};

  @keyframes blinker {
    50% {
      box-shadow: 2px 2px 10px 10px rgb(255, 0, 0, 0.2);
      background: rgb(255, 0, 0, 0.3);
    }
  }

  .status {
    height: 30%;
  }

  .action-section {
    display: flex;
    gap: 15px;
    justify-content: center;
    align-items: center;
    padding-bottom: 15px;
  }
`;

interface InfoSectionProps {
  statusMessage: string;
  isActionRequired: boolean;
  onUpdateClick?: () => void;
  onActClick?: () => void;
  isManual?: boolean;
}

const InfoCard: React.FC<InfoSectionProps> = ({
  statusMessage,
  isActionRequired,
  isManual,
  onActClick,
  onUpdateClick,
}) => {
  return (
    <StyledInfoSection $isAlert={!isManual && isActionRequired}>
      <p className="status">Status: {statusMessage}</p>
      {isActionRequired && (
        <div className="action-section">
          <div>Action Required</div>
          {!isManual && (
            <Button variant="success" onClick={onActClick}>
              Take Action
            </Button>
          )}
        </div>
      )}
      {isManual && <Button onClick={onUpdateClick}>Update Data</Button>}
    </StyledInfoSection>
  );
};

export default InfoCard;
