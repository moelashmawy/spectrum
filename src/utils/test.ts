import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { ReactElement } from "react";

export const createSnapshot = (ComponentToTest: ReactElement) => {
  const {
    container: { firstChild: actual },
  } = render(ComponentToTest);
  expect(actual).toMatchSnapshot();
};
