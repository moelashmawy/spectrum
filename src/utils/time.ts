export const getCurrentTime = () => {
  const d = new Date();
  return d.toLocaleTimeString();
};
