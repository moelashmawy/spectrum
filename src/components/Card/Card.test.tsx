import Card from "./Card";
import { createSnapshot } from "../../utils/test";
import { render, screen } from "@testing-library/react";

describe("Button", () => {
  test("Card snapshot with all data", () => {
    createSnapshot(
      <Card
        statusMessage="Rocket systems are operational and monitoring."
        isActionRequired={false}
        isManual={true}
      />,
    );
  });

  test("Show action required when action is required", () => {
    render(
      <Card
        statusMessage="Rocket systems are operational and monitoring."
        isActionRequired={true}
        isManual={true}
      />,
    );

    const ActionRequired = screen.getByText("Action Required");
    expect(ActionRequired).toBeInTheDocument();
  });

  test("Show Update Data button when mode is manual", () => {
    render(
      <Card
        statusMessage="Rocket systems are operational and monitoring."
        isActionRequired={true}
        isManual={true}
      />,
    );

    const UpdateDataButton = screen.getByText("Update Data");
    expect(UpdateDataButton).toBeInTheDocument();
  });

  test("Show Take Action button when mode is real time and action required", () => {
    render(
      <Card
        statusMessage="Rocket systems are operational and monitoring."
        isActionRequired={true}
        isManual={false}
      />,
    );

    const TakeActionButton = screen.getByText("Take Action");
    expect(TakeActionButton).toBeInTheDocument();
  });
});
