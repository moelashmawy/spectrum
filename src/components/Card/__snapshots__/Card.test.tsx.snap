// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Card snapshot with all data 1`] = `
<div
  class="sc-gsTDqH gbgLXD"
>
  <p
    class="status"
  >
    Status: 
    Rocket systems are operational and monitoring.
  </p>
  <button
    class="sc-bdfCDU fLjGwI"
  >
    Update Data
  </button>
</div>
`;
